functions/snippets/src/main/java/ParseContentType.java

```diff
@@ -16,46 +16,52 @@
 
 // [START functions_http_content]
 
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
 import com.google.gson.Gson;
 import com.google.gson.JsonObject;
+
+import java.io.BufferedWriter;
 import java.io.IOException;
-import java.io.PrintWriter;
+import java.nio.charset.Charset;
 import java.util.Base64;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
 
-public class ParseContentType {
+public class ParseContentType implements HttpFunction {
 
   // Use GSON (https://github.com/google/gson) to parse JSON content.
   private Gson gsonParser = new Gson();
 
   // Responds to an HTTP request using data from the request body parsed according to the
   // "content-type" header.
-  public void parseContentType(HttpServletRequest request, HttpServletResponse response)
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
       throws IOException {
     String name;
-    String contentType = request.getContentType();
+    String contentType = request.getContentType().get();
     if (contentType.equals("application/json")) {
       // '{"name":"John"}'
       JsonObject body = gsonParser.fromJson(request.getReader(), JsonObject.class);
       name = body.get("name").getAsString();
     } else if (contentType.equals("application/octet-stream")) {
       // 'John', stored in a Buffer
-      name = new String(Base64.getDecoder().decode(request.getInputStream().readAllBytes()));
+      name = new String(
+          Base64.getDecoder().decode(request.getInputStream().readAllBytes()),
+          Charset.defaultCharset());
```

Comment:
I believe this is the default behavior for String - why not just use `new String(byte[])`?